reason-react-tutorial.md

```diff
@@ -0,0 +1,681 @@
+
+The first thing we've done inside of our side effect is pull our input value out of `self.state.input`. We'll use this for our API query coming up.
+
+## Data Fetching
+
+We've come a long way! We've got an operating form that triggers our loading state and a `<Card />` component for once we've got a list of results. Now we just need to connect the dots and get the real data from GitHub's API into our app.
+
+Data fetching in Reason is a lot easier said than done. Actually calling the API isn't too hard, but the trickiness starts once we recieve a response. Because Reason is statically typed it needs to make sure that the API response is correctly mapped into valid Reason/OCaml types. We call this process of parsing the JSON and transforming it into valid types **JSON decoding**.
+
+JSON decoding can be kind of tricky. The "proper" way to do it is to declare every single key* in the JSON that you care about. Then you try to map each key to the type you want it to be on the Reason side. If it maps, great! But if it doesn't map correctly you assume it's bad data and throw out the entire key, potentially replacing it with a default value. This can get really verbose, but this method ensures that you handle any malformed data when it enters your app instead of letting it cause bugs later on.
+
+Granted, you could write some [external bindings](https://bucklescript.github.io/docs/en/intro-to-external#special-identity-external) and essentially tell the compiler "this is what my JSON looks like and it will never be different than this type". But rarely in the real world do our external APIs _always_ returns _exactly_ what we expect. Sometimes they crash or return 500 errors. Sometimes that key we expected to contain a number is actually `null`. Cutting corners on type bindings here might be convenient, but one of the main selling points of using a typed language like Reason is the compiler and the safety a it brings to the table.
+
+All that said, since we're doing this tutorial to get a flavor of what ReasonReact feels like, we'll do the full JSON decoding. There's a few community libraries to make our JSON decoding and API fetching a bit easier. So before we jump into our fetching logic, lets install `bs-fetch` and `@glennsl/bs-json`. The first is a thin wrapper around the native `window.fetch` function, and the second will give us a bunch of utility functions to ease the decoding process.
+
+{{< highlight javascript >}}
+npm i bs-fetch @glennsl/bs-json
```

Comment:
Maybe use `npm install` to stay consistent. I always try to not use the aliases to not confuse anyone :)

Reply:
Nice catch! Agreed, that's what I get for always using them myself haha!